lib/galaxy/datatypes/constructive_solid_geometry.py

```diff
@@ -51,9 +51,14 @@ def sniff_prefix(self, file_prefix: FilePrefix):
         The structure of a typical PLY file:
         Header, Vertex List, Face List, (lists of other elements)
         """
-        if not self._is_ply_header(file_prefix.string_io(), self.subtype):
-            return False
-        return True
+        try:
+            if self._is_ply_header(file_prefix.string_io(), self.subtype):
```

Comment:
There's a few odd or wrong patterns in this PR which I think we can improve by adding a [TextIOWrapper](https://docs.python.org/3/library/io.html#io.TextIOWrapper) to the StringIO class and using that with the `errors='ignore'`. Because I think that's essentially the behavior you're going for here, that is dealing with mixed utf-8 and non-utf-8 binary content, where the sniffer just looks at the utf-8 content. I'll push a commit and you can let me know if that is still correct.

Reply:
@mvdbeek  Thanks.   This does require reading the header portions of files that may contain non-header data in either binary or ascii format.   

The sniff_prefix function encountered error from lib/galaxy/datatypes/sniff.py in **class FilePrefix**:  
Is there a way to deal with that more generally in the **class FilePrefix**

> utf-8' codec can't decode byte 0x85 in position 478: invalid start byte

The set_meta functions had the same error when opened as 'r'.  

I'll take a look at TextIOWrapper

Reply:
https://github.com/galaxyproject/galaxy/compare/dev...mvdbeek:constructive_solid_geometry_non_utf8_error?expand=1 is how I would do it. This should tick all your boxes. I did lose your commit authorship unfortunately, but if that looks good to you I would push it here.

Reply:
@mvdbeek No worries. I don't care about authorship, just want things to work as best possible.   Can you go ahead and push it?  Many Thanks. JJ